internal/users/db/sql/create_schema.sql

```diff
@@ -0,0 +1,36 @@
+CREATE TABLE IF NOT EXISTS users (
+	name				TEXT NOT NULL UNIQUE,
+	uid					INTEGER	NOT NULL UNIQUE,
+	gid					INTEGER NOT NULL,
+	gecos				TEXT DEFAULT "",
+	home				TEXT DEFAULT "",
+	shell				TEXT DEFAULT "/bin/bash",
+	last_login 	        DATE,
+	broker_id           TEXT DEFAULT "",
+	PRIMARY KEY(uid)
+);
+CREATE UNIQUE INDEX "idx_user_name" ON users ("name");
```

Comment:
@jibel You mentioned in a call that you would prefer to not use the UNIQUE constraint in the schema, but I did not understand the reasoning. Can you please explain it again, so that I can update the schema and add a code comment with the reasoning?

Reply:
It's mostly for convenience from a database management point of view.
Technically unique index and contrainst are similar and can serve the same purpose. A unique index can enforce data integrity and a unique constraint optimizes data lookup.
However, when it comes to modifying data, that could temporarily alter integrity contrainsts, for instance when batch loading data or mass updating a table, the constraint cannot be temporarily dropped while you can drop and recreate an index. Unfortunately sqlite doesn't support "alter table ... drop contraint".

Reply:
That makes sense, thanks! I added comments explaining that the uniqueness is enforced by the index.